src/app/modules/myinfo/myinfo.format.ts

```diff
@@ -57,26 +57,32 @@ export const formatAddress = (addr: MyInfoAddress | undefined): string => {
   // 4. floor + unit (if available),
   // 5. country
   // 6. postal
-  const buildingBlocks = []
+  const buildingBlocks: string[] = []
 
-  if (building.value) {
+  if (building?.value) {
     buildingBlocks.push(`${building.value},`)
   }
-  buildingBlocks.push(block.value)
-  if (street.value) {
+  if (block?.value) {
+    buildingBlocks.push(block.value)
+  }
+  if (street?.value) {
     buildingBlocks.push(`${street.value},`)
   }
 
-  if (floor.value && unit.value) {
+  if (floor?.value && unit?.value) {
     buildingBlocks.push(`#${floor.value}-${unit.value},`)
   }
 
-  buildingBlocks.push(country.desc)
+  if (country?.desc) {
+    buildingBlocks.push(country.desc)
+  }
 
-  buildingBlocks.push(postal.value)
+  if (postal?.value) {
+    buildingBlocks.push(postal.value)
+  }
 
   // Return string form with each block being separated by a space.
-  return buildingBlocks.filter((b) => b !== '').join(' ')
+  return buildingBlocks.join(' ')
```

Comment:
why the change here?

Reply:
if we check that all the values are truthy before pushing them into the array, then none of them can be empty strings